cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -103,11 +107,111 @@
   private final String instanceName;
   private final String sentryAdminGroup;
 
-  AuthBinding(String sentrySite, String instanceName, String sentryAdminGroup) {
+  private final LoadingCache<Principal, Set<String>> groupCache;
```

Comment:
I don't think that the implementation will change to not a have a cache if it already has one I think we can rely on it. But if it really seems something which we want to be certain of then we can have the cache. 

With inconsistency and overhead I meant what happens with this cache when an user is added or removed from a group in the underlying system. Since the call does not pass through CDAP it has no way to invalidate/refresh the entry so we will have to wait for it to be refreshed on time duration.